tensorflow/lite/kernels/eigen_support.cc

```diff
@@ -130,7 +130,14 @@ RefCountedEigenContext* GetEigenContext(TfLiteContext* context) {
 }
 
 TfLiteStatus Refresh(TfLiteContext* context) {
-  SetEigenNbThreads(context->recommended_num_threads);
+  // if no of thread  < -1 , then there is no point of continuing
```

Comment:
I don't think this is right, what if we want to reset to the "default" case?

Reply:
@jdduke : I think it is better to restrict a default value indication to -1 only. If user want to reset to default, then -1 should be used. I have followed the logic in that manner.

Reply:
Seems like we should fix this in the interpreter to avoid these kinds of cases in the first place?

Reply:
@jdduke : Yes, you are right, there are two ends to handle in these type of problems. Because there are multiple layered operation involved in between Interpreter & EigenThread, so i feel better to have a safe check here as well. But it would be nice to add a LOG before returning. 
I think in Model & Interpreter also, i will add an error check and warning message. In that way we can ensure safety in both the ends.
Please let me know your opinions. Thanks!

Reply:
@jdduke :  I have done some additional modifications based on your concern shared. Please let me know your valuable opinions, Thanks!
Now with new changes i made all external context to have default no of threads as 4.